core/src/modules/locks.ts

```diff
@@ -41,3 +41,35 @@ export async function waitForLock(
 
   return { releaseLock, attempts };
 }
+
+export type getLockResponse = {
+  releaseLock: Function;
+  isLocked: boolean;
+  lockedBy: string;
+};
+
+export async function getLock(
+  key: string,
+  requestId: string = uuid.v4(),
+  ttl = LOCK_DURATION_MS,
+  lockType?: string
+): Promise<getLockResponse> {
+  const client = api.redis.clients.client;
+  const lockKey = `grouparoo:lock:${
+    lockType.length > 1 && `${lockType}:`
+  }${key}`;
+
+  const set = await client.setnx(lockKey, requestId);
+
+  await client.expire(lockKey, Math.ceil(ttl / 1000));
```

Comment:
This changes the expire TTL on the key regardless of this caller created it.  I would guess you don't want to mess with someone else's lock unless set===true